algo/src/test/java/com/hartwig/serve/sources/ckb/ActionableTrialFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.VariantRequirementDetail;
+import com.hartwig.serve.datamodel.EvidenceDirection;
+import com.hartwig.serve.datamodel.EvidenceLevel;
+import com.hartwig.serve.datamodel.Knowledgebase;
+
+import org.apache.commons.compress.utils.Sets;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+public class ActionableTrialFactoryTest {
+
+    @Test
+    public void canCreateActionableTrial() {
+        Location location = ImmutableLocation.builder().nctId("nctid").city("").country("Netherlands").status("Recruiting").build();
+        VariantRequirementDetail requirementType =
+                ImmutableVariantRequirementDetail.builder().profileId(0).requirementType("required").build();
+        CkbEntry entry = CkbTestFactory.createEntryWithClinicalTrial(location, "Recruiting", requirementType);
+        ActionableTrialFactory actionableTrialFactory = new ActionableTrialFactory();
+        Set<ActionableEntry> entrySet = actionableTrialFactory.create(entry, "KRAS", "gene");
+
+        assertEquals(1, entrySet.size());
+        ActionableEntry trial = entrySet.iterator().next();
+        assertEquals(Knowledgebase.CKB_TRIAL, trial.source());
+        assertEquals("KRAS", trial.sourceEvent());
+        assertEquals(Sets.newHashSet("https://clinicaltrials.gov/study/nctid"), trial.sourceUrls());
+        assertEquals("title", trial.treatment().name());
+        assertEquals("AB", trial.applicableCancerType().name());
+        assertEquals("162", trial.applicableCancerType().doid());
+        assertEquals(Sets.newHashSet(), trial.blacklistCancerTypes());
+        assertEquals(EvidenceLevel.B, trial.level());
+        assertEquals(EvidenceDirection.RESPONSIVE, trial.direction());
+        assertEquals(Sets.newHashSet("Netherlands"), trial.evidenceUrls());
+    }
+
+    @Test
+    public void canDetermineCountriesToInclude() {
+        ClinicalTrial trialPotentiallyOpenInBelgium = createTrial("Recruiting", "Belgium", "Recruiting");
+        ClinicalTrial trialNotOpenInNetherlands = createTrial("Recruiting", "Netherlands", "Not yet recruiting");
+        ClinicalTrial trialPotentiallyOpenInUnitedStates = createTrial("Recruiting", "United States", "Recruiting");
+        ClinicalTrial trialWithdrawnInUnitedStates = createTrial("Recruiting", "United States", "Withdrawn");
+        ClinicalTrial trialPotentiallyOpenInBelgiumAndIndia =
+                createTrialWithMultipleLocations("Recruiting", "Belgium", "Recruiting", "India", "Recruiting");
+        ClinicalTrial trialPotentiallyOpenInBelgiumAndGermany =
+                createTrialWithMultipleLocations("Recruiting", "Belgium", "Recruiting", "Germany", "Active, not recruiting");
+        ClinicalTrial trialPotentiallyOpenInBelgiumSuspendedInGermany =
+                createTrialWithMultipleLocations("Recruiting", "Belgium", "Recruiting", "Germany", "Suspended");
+        ClinicalTrial trialPotentiallyOpenInBelgiumSuspendedInIndia =
```

Comment:
I think your test becomes more comprehensive when you make "suspendedInBelgiumOpenInIndia"?

Reply:
Agree, added this test